lib/static/new-ui/features/suites/components/SuitesTreeView/index.tsx

```diff
@@ -0,0 +1,155 @@
+    const navigate = useNavigate();
+    const {suiteId} = useParams();
+
+    const list = useList({
+        items: props.treeViewItems,
+        withExpandedState: true,
+        getItemId: item => {
+            return item.fullTitle;
+        },
+        controlledState: {
+            expandedById: props.treeViewExpandedById
+        }
+    });
+
+    const parentRef = React.useRef<HTMLDivElement>(null);
+
+    const virtualizer = useVirtualizer({
+        count: list.structure.visibleFlattenIds.length,
+        getScrollElement: () => parentRef.current,
+        estimateSize: () => 28,
+        getItemKey: useCallback((index: number) => list.structure.visibleFlattenIds[index], [list]),
+        overscan: 200
+    });
+
+    const virtualizedItems = virtualizer.getVirtualItems();
+
+    // Effects
+    useEffect(() => {
+        if (!suiteId && props.currentSuiteId && suiteId !== props.currentSuiteId) {
+            navigate(encodeURIComponent(props.currentSuiteId));
+        }
+    }, []);
+
+    useEffect(() => {
+        if (!props.isInitialized) {
+            return;
+        }
+
+        props.actions.setStrictMatchFilter(false);
+
+        if (suiteId) {
+            props.actions.suitesPageSetCurrentSuite(suiteId);
+            virtualizer.scrollToIndex(list.structure.visibleFlattenIds.indexOf(suiteId), {align: 'start'});
+        }
+    }, [props.isInitialized]);
+
+    // Event handlers
+    const onItemClick = useCallback(({id}: {id: string}): void => {
+        const item = list.structure.itemsById[id];
+
+        if (item.type === TreeViewItemType.Suite && list.state.expandedById && id in list.state.expandedById && list.state.setExpanded) {
+            props.actions.toggleSuiteSection({suiteId: item.fullTitle, shouldBeOpened: !props.treeViewExpandedById[item.fullTitle]});
+        } else if (item.type === TreeViewItemType.Browser) {
+            props.actions.suitesPageSetCurrentSuite(id);
+
+            navigate(encodeURIComponent(item.fullTitle));
+        }
+    }, [list, props.actions, props.treeViewExpandedById]);
+
+    return <ListContainerView className={styles.treeView}>
+        <div ref={parentRef} className={styles['tree-view__container']}>
+            <div
+                className={styles['tree-view__total-size-wrapper']}
+                style={{height: virtualizer.getTotalSize()}}
+            >
+                <div
+                    className={styles['tree-view__visible-window']}
+                    style={{transform: `translateY(${virtualizedItems[0]?.start ?? 0}px)`}}
+                >
+                    {virtualizedItems.map((virtualRow) => {
+                        const item = list.structure.itemsById[virtualRow.key];
+                        const classes = [styles['tree-view__item']];
+                        if (item.fullTitle === props.currentSuiteId) {
+                            classes.push(styles['tree-view__item--current']);
```

Comment:
It's not applicable in this case. Resulting class name is not `tree-view__item--current`. It's `snPAds6U3DI9gceMfgWR`, because we are using CSS modules.

I've re-written this code to better utilise `classnames` library instead:
```
const classes = [
    styles['tree-view__item'],
    {
        [styles['tree-view__item--current']]: item.fullTitle === props.currentSuiteId,
        [styles['tree-view__item--browser']]: item.type === TreeViewItemType.Browser,
        [styles['tree-view__item--error']]: item.type === TreeViewItemType.Browser && (item.status === 'fail' || item.status === 'error')
    }
];
```